info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java

```diff
@@ -320,11 +321,14 @@ void storeElectionContestInDatabase(String electionQueryId, JsonObject contest)
     List<Value<Boolean>> candidateIncumbency =
         new ArrayList<>(electionEntity.getList("candidateIncumbency"));
     // Obtain candidate information and create candidate entities in the database.
+    String state = electionEntity.getString("state");
+    String stateName = state.isEmpty() ? null : State.abbreviationToName.get(state);
```

Comment:
Is it possible for state to be invalid or lower case? In that case your map lookup would fail. If so, it would be good to convert state to uppercase before doing the lookup

Reply:
The state is guaranteed to be two upper class letters or an empty string when it gets stored into the database (implementation can be viewed [here](https://github.com/googleinterns/voter-central/blob/kathy_extend_RelevancyChecker/info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java#L284)). In InfoCompilerTest, `storeBaseElectionInDatabase_checkDatastoreEntityConstructionFromJsonWithState` and `storeBaseElectionInDatabase_checkDatastoreEntityConstructionFromJsonWithoutState` will check for that